From 741f960d15355fabdcecd6810440bad945d1b88f Mon Sep 17 00:00:00 2001
From: Nicolin Chen <b42378@freescale.com>
Date: Thu, 29 Aug 2013 16:35:46 +0800
Subject: [PATCH 411/505] ENGR00276567-2 mxc: tuner: Add si476x driver support

The frequence setting for si476x and old si4763 drivers are a bit different.
Thus add checking code to detect which driver is being used.

Acked-by: Wang Shengjiu <b02247@freescale.com>
Signed-off-by: Nicolin Chen <b42378@freescale.com>
---
 test/mxc_tuner_test/mxc_tuner_test.c  | 20 +++++++++++++++++---
 test/mxc_tuner_test/mxc_tuner_test.sh |  2 +-
 2 files changed, 18 insertions(+), 4 deletions(-)

diff --git a/test/mxc_tuner_test/mxc_tuner_test.c b/test/mxc_tuner_test/mxc_tuner_test.c
index e54bf33..4c9794b 100644
--- a/test/mxc_tuner_test/mxc_tuner_test.c
+++ b/test/mxc_tuner_test/mxc_tuner_test.c
@@ -19,6 +19,7 @@
 #include <unistd.h>
 #include <string.h>
 #include <sys/ioctl.h>
+#include <sys/stat.h>
 #include <linux/videodev2.h>
 
 #define TRIES		25		/* Get 25 samples */
@@ -34,10 +35,17 @@
 
 int main(int argc, char **argv)
 {
-	int fd, ret, n_selection, freq;
+	int fd, ret, n_selection, freq, core;
 	struct v4l2_frequency vt;
+	struct stat buf;
 	char *dev = NULL;
 
+	/* Check if using si476x drivers, return 0 means true */
+	core = stat("/sys/bus/i2c/drivers/si476x-core", &buf);
+	printf("Using si476%c drivers.....\n", core ? '3' : 'x');
+
+	memset(&vt, 0, sizeof(struct v4l2_frequency));
+
 	do{
 		printf("Welcome to radio menu\n");
 		printf("1.-Turn on the radio\n");
@@ -65,7 +73,10 @@ int main(int argc, char **argv)
 				printf("error: Turn on the radio first\n");
 				break;
 			}
-			printf("Frequency = %d\n", vt.frequency);
+			if (!core)
+				printf("Frequency = %d\n", vt.frequency * 625 / 100000);
+			else
+				printf("Frequency = %d\n", vt.frequency);
 			break;
 
 		case TUNER_SETFREQ:
@@ -76,7 +87,10 @@ int main(int argc, char **argv)
 			}
 			printf("Set frequency = ");
 			if (scanf("%d", &freq) == 1){
-				vt.frequency = freq;
+				if (!core)
+					vt.frequency = freq * 100000 / 625;
+				else
+					vt.frequency = freq;
 				ret = ioctl(fd, VIDIOC_S_FREQUENCY, &vt);
 				if (ret < 0) {
 					perror("error: Turn on the radio first\n");
diff --git a/test/mxc_tuner_test/mxc_tuner_test.sh b/test/mxc_tuner_test/mxc_tuner_test.sh
index 9dd11ae..6599b4b 100644
--- a/test/mxc_tuner_test/mxc_tuner_test.sh
+++ b/test/mxc_tuner_test/mxc_tuner_test.sh
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-num=`arecord -l |grep -i "si4763" |awk '{ print $2 }'|sed 's/://'`
+num=`arecord -l |grep -i "si476" |awk '{ print $2 }'|sed 's/://'`
 
 arecord -D hw:$num,0 -f dat | aplay -D hw:0,0 -f dat &
 ./mxc_tuner_test.out
-- 
1.8.0

